siriuspy/siriuspy/devices/dvf.py

```diff
@@ -385,22 +400,45 @@ def cmd_reset(self, timeout=None):
         self.acquisition_time = params.ACQUISITION_TIME_DEFAULT
         return True
 
-    def cmd_acquire_on(self, timeout=None):
+    def cmd_acquire_on(self, timeout=_default_timeout):
         """Tune IOC image acquisition on."""
         return self._set_and_wait('cam1:Acquire', 1, timeout=timeout)
 
-    def cmd_acquire_off(self, timeout=None):
+    def cmd_acquire_off(self, timeout=_default_timeout):
         """Tune IOC image acquisition off."""
         return self._set_and_wait('cam1:Acquire', 0, timeout=timeout)
 
-    def cmd_set_cam_roi(self, offsetx, offsety, width, height, timeout=None):
+    def cmd_cam_roi_set(self, offsetx, offsety, width, height, timeout=None):
         """Set cam image region of interest and reset aquisition."""
+        offsetx, offsety = int(offsetx), int(offsety)
+        width, height = int(width), int(height)
+        if offsetx + width > self.cam_max_sizex:
+            raise ValueError('Invalid offsetx and width combination!')
+        if offsety + height > self.cam_max_sizey:
+            raise ValueError('Invalid offsety and height combination!')
+
         self.cmd_acquire_off(timeout=timeout)
-        self.cam_offsetx = offsetx
-        self.cam_offsety = offsety
-        self.cam_width = width
-        self.cam_height = height
+        if width < self.cam_width:
+            self._set_and_wait('cam1:Width', width, timeout=timeout)
+            self._set_and_wait('cam1:OffsetX', offsetx, timeout=timeout)
+        else:
+            self._set_and_wait('cam1:OffsetX', offsetx, timeout=timeout)
+            self._set_and_wait('cam1:Width', width, timeout=timeout)
```

Comment:
I didn't know the order of the operations was important depending on the size of the region.

What happens if we run them swapped?

Reply:
Got it. I'll investigate why this might be happening. Good to know about this issue.